src/core/proj/qgsprojutils.cpp

```diff
@@ -203,6 +203,31 @@ QgsProjUtils::proj_pj_unique_ptr QgsProjUtils::crsToSingleCrs( const PJ *crs )
       return res;
     }
 
+    case PJ_TYPE_VERTICAL_CRS:
+      return nullptr;
+
+    // maybe other types to handle??
+
+    default:
+      return crsToSingleCrs( crs );
+  }
+
+#ifndef _MSC_VER  // unreachable
+  return nullptr;
+#endif
+}
+
+QgsProjUtils::proj_pj_unique_ptr QgsProjUtils::crsToSingleCrs( const PJ *crs )
```

Comment:
nitpicking: in ISO-19111 a single CRS is the opposite of a compound CRS. But here if you call crsToSingleCrs() on a compound CRS, it will return a clone of itself, which is not a single CRS.

Maybe this method should be called unboundCrs() instead ?

Reply:
Good idea, done